src/context/ShoppingCartContext.tsx

```diff
@@ -0,0 +1,100 @@
+import { createContext, ReactNode, useContext, useState } from "react"
+import { ShoppingCart } from "../components/ShoppingCart"
+import { useLocalStorage } from "../hooks/useLocalStorage"
+
+type ShoppingCartProviderProps = {
+  children: ReactNode
+}
+
+type CartItem = {
+  id: number
+  quantity: number
+}
+
+type ShoppingCartContext = {
+  openCart: () => void
+  closeCart: () => void
+  getItemQuantity: (id: number) => number
+  increaseCartQuantity: (id: number) => void
+  decreaseCartQuantity: (id: number) => void
+  removeFromCart: (id: number) => void
+  cartQuantity: number
+  cartItems: CartItem[]
+}
+
+const ShoppingCartContext = createContext({} as ShoppingCartContext)
+
+export function useShoppingCart() {
+  return useContext(ShoppingCartContext)
```

Comment:
Directly returning the value here is not right, try something like this

```tsx
  const value =useContext(ShoppingCartContext);
  if (!value)
    throw new Error(
      "useShoppingCartContext: undefined value. ShoppingCartProvider must be a parent of this component."
    );
  return value;
```

Reply:
Resolved